Source/CarthageKit/Xcode.swift

```diff
@@ -821,12 +832,20 @@ public func buildInDirectory( // swiftlint:disable:this function_body_length
 				guard let dependency = dependency else {
 					return .empty
 				}
-
-				return createVersionFile(
-					for: dependency.dependency, version: dependency.version,
-					platforms: options.platforms, buildProducts: urls, rootDirectoryURL: rootDirectoryURL
-				)
-				.flatMapError { _ in .empty }
+				var copyToSharedCache = SignalProducer<(URL), CarthageError>.empty
+				if options.cacheBuilds {
+					let sharedCacheURLForDependency = sharedCacheURLByXcodeVersionFor(dependency: dependency.dependency, pinnedVersion: dependency.version)
+					copyToSharedCache = copyFrameworksAndCreateVersionFile(urls, from: rootDirectoryURL, to: sharedCacheURLForDependency,
```

Comment:
I would split this function in 2. One that copies the frameworks and one that creates the version file. Is there any practical reason why they are combined?

Reply:
My refactoring make me using it twice, in https://github.com/openium/Carthage/blob/master/Source/CarthageKit/Xcode.swift#L838 and https://github.com/openium/Carthage/blob/master/Source/CarthageKit/Project.swift#L1400
So to not duplicate code as now copy can be from DerivedData to Carthage/Build and then from CarthageBuild to ~/.../SharedCache (see paths in https://github.com/Carthage/Carthage/issues/2400#issuecomment-427343658)

Reply:
ok